### PR TITLE
Fix select component

### PR DESCRIPTION
**Pull request by Hyperkid123**
@chalettu Stumbled upon issue with multiple versions of react whiles using select component. 

This is caused by react-select v1. It is using old version of react and old API for refs. That blows up ui in Miq.

### CODE REVIEW

**Comment by Hyperkid123** [comment]
@miq-bot assign @karelhala 
@miq-bot add_label bug

File: src/forms/finalFormComponent.jsx [hunk 1]
```diff
@@ -5,25 +5,34 @@ import PropTypes from 'prop-types';
...
+    options={rest.options.filter(option => option.hasOwnProperty('value') && option.value !== null)} // eslint-disable-line
+    value={rest.options.filter(({ value }) => (rest.multi ? input.value.includes(value) : value === input.value))}
+    isMulti={rest.multi}
+    isSearchable={!!rest.searchable}
+    isClearable={!!rest.clearable}
+    hideSelectedOptions={false}
+    closeMenuOnSelect={!rest.multi}
+    noOptionsMessage={() => __('No option found')}
+    onChange={option =>
+    input.onChange(rest.multi ? selectValue(option) : option ? option.value : undefined)} // eslint-disable-line no-nested-ternary
```
Comment (karelhala) [inline src/forms/finalFormComponent.jsx]: Please align this line and use shortcut (if applicable)
```
onChange={
  option => input.onChange(rest.multi ? selectValue(option) : option && option.value)
}
```

**Comment by Hyperkid123** [comment]
@karelhala do you have any idea why the tests are busted again? It works locally...

**Comment by Hyperkid123** [comment]
@karelhala oh its the tagging component.
@PanSpagetka do you mind if i remove the snapshot test? Travis can't handle that.

**Comment by karelhala** [comment]
I see a lot of outdated snapshots in the build.

**Comment by Hyperkid123** [comment]
i get why most of them changed. The select component was used in them, but the tagging one has over 80k lines.

**Comment by PanSpagetka** [comment]
@Hyperkid123 It is ok for me if you remove snapshots

**Comment by Hyperkid123** [comment]
@karelhala ok i've removed a lot of snaphost, something had to change in some dependency, because the snapshots were incredibly bloated. I've replaced them with shallow snapshots...

**Comment by chalettu** [comment]
@karelhala , will merging this PR go ahead and bump the react-ui-components version number for the package in NPM?   I am dependent on this fix for a PR we have been working on

**Comment by karelhala** [comment]
@chalettu yes, new version of `react-ui-components` is automatically released with new version `3.23.1`

**Comment by Hyperkid123** [comment]
@chalettu it should release new version in a second

**Pull request merged**